### PR TITLE
Allow static members in interfaces to treat type parameters as invariant

### PR DESCRIPTION
**Pull request by AlekseyTs**
https://github.com/dotnet/csharplang/issues/3275
https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-06-24.md#interface-static-member-variance
https://github.com/dotnet/csharplang/blob/master/proposals/variance-safety-for-static-interface-members.md

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @RikkiGibson, @cston, @jcouv, @dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@333fred, @RikkiGibson, @cston, @jcouv, @dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Portable/Errors/ErrorCode.cs [hunk 1]
```diff
@@ -1866,6 +1866,7 @@ internal enum ErrorCode
         ERR_StaticAPIInRecord = 8877,
         ERR_CopyConstructorWrongAccessibility = 8878,
         ERR_NonPrivateAPIInRecord = 8879,
+        ERR_UnexpectedVarianceStaticMemmber = 8880,
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Errors/ErrorCode.cs:1869]: nit: Memmber -> Member

**Comment by CyrusNajmabadi** [comment]
IDE side is fine :)

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 2]
```diff
@@ -59246,6 +59246,168 @@ interface I1<out T1, in T2>
...
+    static T1 M1(T1 x) => x;
+    static T2 M2(T2 x) => x;
+}
+";
+
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugExe,
+                                                 parseOptions: TestOptions.Regular8,
+                                                 targetFramework: TargetFramework.NetStandardLatest);
+            compilation1.VerifyDiagnostics(
+                // (13,18): error CS8880: Invalid variance: The type parameter 'T1' must be contravariantly valid on 'I2<T1, T2>.M1(T1)' unless language version 'preview' or greater is used. 'T1' is covariant.
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs:59322]: >must be contravariantly valid [](start = 84, length = 29)

I think a better wording would be "cannot be used contravariantly".

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs:59322]: > I think a better wording would be "cannot be used contra/co/invariantly".

I am sharing the wording with existing message. Changing the wording is outside of the scope of this PR

---
In reply to: [458353056](https://github.com/dotnet/roslyn/pull/46141#discussion_r458353056) [](ancestors = 458353056)

**Comment by AlekseyTs** [comment]
@cston, @jcouv Please review a relatively small fix, need a sign-off from a senior dev.

**Pull request merged**